app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,138 @@
+            'course_outline': 'The outline of the course: ',
+            'grading_policy': 'The grading policy: ',
+            'class_policy': 'The class policy: ',
+            'customization': 'Additional customization notes: '
+        }
+
+        self.model = model
+        self.prompt_template = prompt_template
+        self.parser = parser
+        self.verbose = verbose
+
+        self.max_attempts = 5
+    
+    def compile_context(self):
+        context = 'Course Information.\n'
+        for ord in self.order:
+            entry = self.attributes.get(ord)
+            if entry is not None:
+                entry = entry.rstrip()
+                if entry[-1] != '.':
+                    entry += '.'
+                context += self.fields_dict[ord] + entry + '\n'
+        return context
+    
+    def validate_response(self) -> bool:
+        try:
+            # Use Pydantic model to validate the response
+            SyllabusFormat(**self.response)
+            return True
+        except ValidationError as e:
+            if self.verbose:
+                logger.error(f"Validation error during Syllabus response validation: {e}")
+            return False
+        except TypeError as e:
+            if self.verbose:
+                logger.error(f"TypeError during Syllabus response validation: {e}")
+            return False
+    
+    def create_syllabus(self):
+        attempts = 0
+        syllabus = None
+        context = self.compile_context()
+        prompt = PromptTemplate(
+            template=self.prompt_template,
+            input_variables=["context"],
+            partial_variables={"format_instructions": self.parser.get_format_instructions()}
+        )
+        chain = prompt | self.model
+
+        while attempts < self.max_attempts:
+            attempts += 1
+            response = chain.invoke({"context": context})
+            if self.verbose:
+                logger.info(f"Generated Syllabus response attempt {attempts}: {response}")
+            try:
+                self.response = self.parser.parse(response)
+                if self.validate_response():
+                    return self.response
+            except (json.JSONDecodeError, OutputParserException) as e:
+                # Log the error or handle it accordingly
+                logger.error(f"Error parsing response: {e}")
+                    
+        # Log if fewer questions are generated
+        if syllabus is None:
+            logger.warning(f"Syllabus was not created")
+
+        return syllabus
+    
+class SyllabusBuilder:
+    def __init__(self, attributes, model = None, prompt_template = None, parser = None, verbose=False):
+        self.verbose = verbose
+        self.attributes = attributes
+
+        default_config = {
+            "model": GoogleGenerativeAI(model="gemini-1.0-pro", temperature=0.8),
+            "prompt_template": read_text_file("prompts/syllabus_prompt.txt"),
```

Comment:
Always prefix file paths with an `r` when importing files to prevent issues with `/` across different platforms:

```python
"prompt_template": read_text_file(r"prompts/syllabus_prompt.txt")
```

Reply:
fixed it